src/nools/lib.js

```diff
@@ -139,7 +156,7 @@ function emitTasksForSchedule(c, schedule, r) {
         _id: (r ? r._id : c.contact && c.contact._id) + '~' + (event.id || i) + '~' + (schedule.name || schedule.index),
         deleted: !!((c.contact && c.contact.deleted) || r ? r.deleted : false),
         doc: c,
-        contact: c.contact,
+        contact: obtainContactFromSchedule(schedule, c, r),
```

Comment:
The emitted task actually has two contact properties. A task looks like this: 

```
    define Task {
      _id: null,
      deleted: null,
      doc: null,
      contact: null,
      icon: null,
      date: null,
      title: null,
      fields: null,
      resolved: null,
      priority: null,
      priorityLabel: null,
      reports: null,
      actions: null
    }
```

The contact property we are altering here is the `Task.content` which controls only what appears on in the Tasks Tab UI. It would be much cleaner if the task just had a `heading` attribute instead of having a contact attribute, because the only thing it is used for is [here](https://github.com/medic/medic/blob/master/webapp/src/js/services/live-list.js#L235).  Hopefully this explains why the contact that is set needs to be formatted as `contact: { name: 'blah' }`.

We could change this in the declarative config schema to be an attribute `heading`, but it would still need to set `contact.name` to be backwards compatible with webapp.

Reply:
Oh, ok, cool! Thanks for explaining. I got the wrong impression from the issue description.